Source/ApiTemplate/Source/ApiTemplate/ApplicationBuilderExtensions.cs

```diff
@@ -43,15 +44,14 @@ public static IApplicationBuilder UseStaticFilesWithCacheControl(this IApplicati
         /// Uses custom serilog request logging. Adds additional properties to each log.
         /// See https://github.com/serilog/serilog-aspnetcore.
         /// </summary>
-        public static IApplicationBuilder UseCustomSerilogRequestLogging(
-            this IApplicationBuilder application,
-            IActionContextAccessor actionContextAccessor) =>
+        public static IApplicationBuilder UseCustomSerilogRequestLogging(this IApplicationBuilder application) =>
             application.UseSerilogRequestLogging(
                 options => options.EnrichDiagnosticContext = (diagnosticContext, httpContext) =>
                 {
-                    var actionDescriptor = actionContextAccessor.ActionContext.ActionDescriptor;
-                    diagnosticContext.Set("ActionName", actionDescriptor.DisplayName);
-                    diagnosticContext.Set("ActionId", actionDescriptor.Id);
+                    var endpoint = httpContext.GetEndpoint();
+                    var routeName = endpoint?.Metadata?.GetMetadata<RouteNameMetadata>()?.RouteName;
```

Comment:
Not that it probably matters but should this be `IRouteNameMetadata`?

Reply:
Good idea, fixed.